background.js

```diff
@@ -28,42 +33,57 @@ function onError(err) {
 
 browser.contextMenus.create({
   id: "twitter-img",
-  title: "twitter image helper",
+  title: "Twitter Image Helper",
   documentUrlPatterns: ["*://*.twitter.com/*"],
 }, onCreated);
 
 browser.contextMenus.create({
   id: "twitter-img-open",
-  title: "Open original (tab)",
+  title: "Open Original (tab)",
   parentId: "twitter-img",
 }, onCreated);
 
 browser.contextMenus.create({
   id: "twitter-img-open-inplace",
-  title: "Open original",
+  title: "Open Original",
+  parentId: "twitter-img",
+}, onCreated);
+
+browser.contextMenus.create({
+  id: "twitter-img-download",
+  title: "Download Original",
   parentId: "twitter-img",
 }, onCreated);
 
 
 browser.contextMenus.onClicked.addListener(function(info, tab) {
-  if(lastOrigUrl === "") {
+  if(lastOrigUrl === "" || fileName === "") {
     // Indicates the right click menu has been 'cleared' by clicking on a non-recognized thing
     return;
   }
-  if(lastOrigUrl === null) {
+  if(lastOrigUrl === null || fileName === null) {
     console.log(`twitter-image-helper: unexpected context menu event with null url: ${info}`);
     return;
   }
   switch (info.menuItemId) {
     case "twitter-img-open":
       browser.tabs.create({
         "url": lastOrigUrl,
+		"active" : false
```

Comment:
style nit, this should be spaces, not tabs, and should have a trailing comma (even though it's optional).

There's another two lines below that also are using tabs instead of spaces.

Reply:
Fixed